storage/pebble/registers_snapshot_reader.go

```diff
@@ -0,0 +1,63 @@
+package pebble
+
+import (
+	"errors"
+
+	"github.com/onflow/flow-go/fvm/storage/snapshot"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RegisterSnapshotReader provides the default implementation of the
+// RegisterSnapshotReader interface.
+//
+// It wraps a storage.RegisterIndexReader and adds logic to construct
+// register snapshots at a given block height. The reader enforces explicit
+// range checks before creating a snapshot, ensuring that queries outside of
+// the indexed range return a clear storage.ErrHeightNotIndexed error.
+//
+// This design improves user experience by preventing confusing error
+// propagation from deeper layers (such as FVM or Cadence) and ensures that
+// missing register values are handled gracefully by returning nil instead
+// of an error.
```

Comment:
this is an implementation detail I don't think we need in the type docs
```suggestion
// propagation from deeper layers (such as FVM or Cadence).
```